src/fireedge/src/modules/components/Forms/VmTemplate/CreateForm/Steps/ExtraConfiguration/context/userInputsSchema.js

```diff
@@ -0,0 +1,286 @@
+          userInputTypes.rangeFloat,
+          userInputTypes.list,
+        ]
+        break
+      default:
+        break
+    }
+
+    return arrayToOptions(defaultValues, {
+      addEmpty: false,
+      getText: (type) => sentenceCase(type),
+    })
+  },
+  validation: string()
+    .trim()
+    .required()
+    .oneOf(valuesOfUITypes)
+    .default(() => valuesOfUITypes[0]),
+  grid: { sm: 6, md: 4 },
+}
+
+/** @type {Field} Description field */
+const DESCRIPTION = {
+  name: 'label',
+  label: T.Description,
+  type: INPUT_TYPES.TEXT,
+  validation: string()
+    .trim()
+    .notRequired()
+    .default(() => undefined),
+  grid: { sm: 6, md: 4 },
+}
+
+/** @type {Field} Options field */
+const OPTIONS = {
+  name: 'options',
+  label: T.Options,
+  tooltip: [T.PressKeysToAddAValue, ['ENTER']],
+  dependOf: TYPE.name,
+  type: INPUT_TYPES.AUTOCOMPLETE,
+  multiple: true,
+  htmlType: (type) =>
+    ![uiList, uiListMultiple].includes(type) && INPUT_TYPES.HIDDEN,
+  validation: array(string().trim())
+    .default(() => [])
+    .when(TYPE.name, (type, schema) =>
+      [uiList, uiListMultiple].includes(type)
+        ? schema.required().min(1)
+        : schema.strip().notRequired()
+    ),
+  fieldProps: {
+    freeSolo: true,
+    placeholder: 'optA,optB,optC',
+  },
+  grid: { md: 8 },
+}
+
+/** @type {{ MIN: Field, MAX: Field }} Range fields */
+const { MIN, MAX } = (() => {
+  const validation = number()
+    .min(0)
+    .default(() => undefined)
+    .when(TYPE.name, (type, schema) =>
+      [uiRange, uiRangeFloat].includes(type)
+        ? schema.required()
+        : schema.strip().notRequired()
+    )
+
+  const common = {
+    dependOf: TYPE.name,
+    type: INPUT_TYPES.TEXT,
+    htmlType: (type) =>
+      [uiRange, uiRangeFloat].includes(type) ? 'number' : INPUT_TYPES.HIDDEN,
+    grid: { sm: 6, md: 4 },
+    fieldProps: (type) => ({ step: type === uiRangeFloat ? 0.01 : 1 }),
+  }
+
+  return {
+    MIN: {
+      ...common,
+      name: 'min',
+      label: T.Min,
+      validation: validation.lessThan(ref('max')),
+    },
+    MAX: {
+      ...common,
+      name: 'max',
+      label: T.Max,
+      validation: validation.moreThan(ref('min')),
+    },
+  }
+})()
+
+/** @type {Field} Default value field */
+const DEFAULT_VALUE = {
+  name: 'default',
+  label: T.DefaultValue,
+  dependOf: [TYPE.name, OPTIONS.name],
+  optionsOnly: true,
+  type: ([type] = []) =>
+    [uiBoolean, uiList, uiListMultiple].includes(type)
+      ? INPUT_TYPES.AUTOCOMPLETE
+      : INPUT_TYPES.TEXT,
+  htmlType: ([type] = []) =>
+    ({
+      [uiNumber]: 'number',
+      [uiNumberFloat]: 'number',
+      [uiPassword]: INPUT_TYPES.HIDDEN,
+    }[type]),
+  multiple: ([type] = []) => type === uiListMultiple,
+  values: ([type, options = []] = []) =>
+    type === uiBoolean
+      ? arrayToOptions(['NO', 'YES'])
+      : arrayToOptions(options),
+  validation: mixed()
+    .default(() => undefined)
+    .when(
+      [TYPE.name, OPTIONS.name],
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).